core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1376,6 +1377,25 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
   private[zio] def tellInterrupt(cause: Cause[Nothing]): Unit =
     tell(FiberMessage.InterruptSignal(cause))
 
+  private[zio] def transferChildren(scope: FiberScope): Unit = {
+    val children = _children
```

Comment:
@kyri-petrou this conflicts with pending work in #8819,
doesn't matter which one gets merged first, this have to executed by the fiber itself.

btw, I think even with the current implementation this misses the `Sync` block protecting the children set